app/models/manageiq/providers/hawkular/middleware_manager/middleware_diagnostic_report.rb

```diff
@@ -0,0 +1,139 @@
+require 'timeout'
+
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager::MiddlewareDiagnosticReport < ApplicationRecord
+    self.table_name = 'middleware_diagnostic_reports'
+
+    STATUS_QUEUED = 'Queued'.freeze
+    STATUS_RUNNING = 'Running'.freeze
+    STATUS_ERROR = 'Error'.freeze
+    STATUS_READY = 'Ready'.freeze
```

Comment:
We could use [rails enums](http://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html) to simplify the logic.

Reply:
Well, we could either leave as it or change to use rails enums. It's your choice now :)
We might as well implement (or use) something on the future that supports string enums.
@mtho11 @abonas WDYT?

Reply:
I'm a fan of enums!